be/src/connector_sink/iceberg_chunk_sink.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceberg_chunk_sink.h"
+
+#include <future>
+
+#include "column/datum.h"
+#include "exec/pipeline/fragment_context.h"
+#include "exprs/expr.h"
+#include "formats/orc/orc_file_writer.h"
+#include "formats/parquet/parquet_file_writer.h"
+#include "formats/utils.h"
+#include "util/url_coding.h"
+#include "utils.h"
+
+namespace starrocks::connector {
+
+IcebergChunkSink::IcebergChunkSink(std::vector<std::string> partition_columns,
+                                   std::vector<std::unique_ptr<ColumnEvaluator>>&& partition_column_evaluators,
+                                   std::unique_ptr<LocationProvider> location_provider,
+                                   std::unique_ptr<formats::FileWriterFactory> file_writer_factory,
+                                   int64_t max_file_size, RuntimeState* state)
+        : _partition_column_names(std::move(partition_columns)),
+          _partition_column_evaluators(std::move(partition_column_evaluators)),
+          _location_provider(std::move(location_provider)),
+          _file_writer_factory(std::move(file_writer_factory)),
+          _max_file_size(max_file_size),
+          _state(state) {}
+
+Status IcebergChunkSink::init() {
+    RETURN_IF_ERROR(ColumnEvaluator::init(_partition_column_evaluators));
+    return Status::OK();
+}
+
+// requires that input chunk belongs to a single partition (see LocalKeyPartitionExchange)
+StatusOr<ConnectorChunkSink::Futures> IcebergChunkSink::add(ChunkPtr chunk) {
+    std::string partition;
```

Comment:
Most of the codes here for three `chunk_sink` are duplicate. Should avoid that

Reply:
The design is to let every connector have its own implementation.